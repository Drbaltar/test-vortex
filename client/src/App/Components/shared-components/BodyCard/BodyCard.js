import React from 'react';

const BodyCard = (props) => {
    const { title } = props;

    let bodyContents, footerContents;

    if (Array.isArray(props.children)) {
        bodyContents = props.children[0];
        footerContents = props.children.length > 1 ? props.children[1] : null;
    } else {
        bodyContents = props.children;
        footerContents = null;
    }


    return (
        <form className="card bg-light" noValidate>
            {getCardHeader(title)}
            {getCardBody(bodyContents)}
            {footerContents ? getCardFooter(footerContents) : null}
        </form>
    );
};

const getCardHeader = (title) => {
    return <h1 className="card-header">{title}</h1>;
};

const getCardBody = (bodyContents) => {
    return (
        <div className="p-4">
            {bodyContents}
        </div>
    );
};

const getCardFooter = (footerContents) => {
    return (
        <div className='card-footer'>
            {footerContents}
        </div>
    );
};

export default BodyCard;
