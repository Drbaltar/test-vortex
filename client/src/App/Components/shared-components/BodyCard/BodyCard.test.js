import React from 'react';
import { shallow } from 'enzyme';
import BodyCard from './BodyCard';

class TestBodyComponent extends React.Component {
    constructor(props) {
        super(props);
    }

    render() { <div />; }
}

class TestFooterComponent extends React.Component {
    constructor(props) {
        super(props);
    }

    render() { <div />; }
}

const testTitle = 'Approve Test Question';

describe('BodyCard', () => {
    const wrapper = shallow(<BodyCard title={testTitle}>
        <TestBodyComponent />
        <TestFooterComponent />
    </BodyCard>);

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('sets the header to the passed in title prop', () => {
        expect(wrapper.find('.card-header').text()).toEqual(testTitle);
    });

    it('renders the child body components in the body of the card', () => {
        expect(wrapper.find('TestBodyComponent').exists()).toBe(true);
    });

    it('renders the child footer component', () => {
        expect(wrapper.find('TestFooterComponent').exists()).toBe(true);

    });

    it('renders the child footer components in the footer of the card', () => {
        expect(wrapper.find('.card-footer').children().name()).toEqual('TestFooterComponent');
    });

    describe('when only one component is passed as a child', () => {
        let singleComponentBodyCard;

        beforeAll(() => {
            singleComponentBodyCard = shallow(<BodyCard title={testTitle}>
                <TestBodyComponent />
            </BodyCard>);
        });

        it('renders the component in the BodyCard body', () => {
            expect(singleComponentBodyCard.find('.p-4').children().name()).toEqual('TestBodyComponent');
        });

        it('does not render anything in the the child footer component', () => {
            expect(singleComponentBodyCard.find('.card-footer').exists()).toBe(false);
        });
    });
});