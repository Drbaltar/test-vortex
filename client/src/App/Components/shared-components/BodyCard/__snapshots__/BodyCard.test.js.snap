// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyCard renders correctly 1`] = `
<form
  className="card bg-light"
  noValidate={true}
>
  <h1
    className="card-header"
  >
    Approve Test Question
  </h1>
  <div
    className="p-4"
  >
    <TestBodyComponent />
  </div>
  <div
    className="card-footer"
  >
    <TestFooterComponent />
  </div>
</form>
`;
